csrc/optimization/move_split_cat.cpp

```diff
@@ -122,14 +231,30 @@ TensorView* findCancelingSplit(CatOp* cat) {
 } // namespace
 
 void MoveSplitCatPass::runPass(Fusion* fusion) {
-  const auto& exprs = fusion->exprs();
-  for (CatOp* cat : ir_utils::filterByType<CatOp>(exprs)) {
-    TensorView* split_in = findCancelingSplit(cat);
+  std::vector<Expr*> exprs = fusion->exprs();
+  for (auto* cat : ir_utils::filterByType<CatOp>(exprs)) {
+    std::vector<Expr*> use_def_chain;
+    TensorView* split_in = findCancelingSplit(cat, std::ref(use_def_chain));
     if (split_in == nullptr) {
       continue;
     }
+
+    TensorView* merged_out = split_in;
+    for (auto i = use_def_chain.rbegin(), end = use_def_chain.rend(); i != end;
+         i++) {
+      Expr* to_replay = *i;
+      if (to_replay->isA<LoadStoreOp>()) {
+        auto* set_out = to_replay->output(0)->as<TensorView>();
+        std::vector<int64_t> permutation = *ir_utils::computePermutation(
+            set_out->getRootDomain(), set_out->getMaybeRFactorDomain());
+        merged_out = permute(merged_out, permutation);
```

Comment:
Honest Q: would something like this be better or worse? This will transfer definitions as you propagate forward, which means undefining the old values, so if you do this you should double check that there are no additional uses dangling around. Note that you don't necessarily need to use `transferDefinitionToNewOutputs` either: if you look at that function it shows how to use `to_replay->newObjectFunc()` to create another definition that would work. The key is to build the output TensorView properly.

```suggestion
        old_out = to_replay->output(0)->as<TensorView>();
        std::vector<IterDomain*> new_root;
        new_root.reserve(old_out->getRootDomain().size());
        for (auto* id : old_out->getRootDomain()) {
          new_root.push_back(
            // You'll need to initialize ca_map earlier and find the original axis from the split so that you can map it here
            ca_map.areMapped(id, original_split_axis, IdMappingMode::EXACT) ?
            IterDomainBuilder(id).extent(catted_extent).build() :
            id);          
        }
        auto* new_out = IrBuilder::create<TensorView>(new_root, old_out->contiguity());
        ir_utils::transferDefinitionToNewOutputs(to_replay, {new_out});
        merged_out = new_out;
```